### ISSUE TITLE
move to more *nix documentation conventions

### ISSUE THREAD

**Issue by johnnybubonic** [unbit/uwsgi-docs#374]
So, currently this is the case:

```
[root@archdev ~]# /tmp/uwsgi --help 2>&1 | wc -l                                                                                                                                                                   
1003
```

This is, quite frankly, overkill. Imagine trying to visually parse that on a VGA TTY. Heck, even ls(1) only has 115 lines of --help output.

What I'd suggest is only list 5-10 of the *most commonly* used options in the --help output, and document the rest in a man page or GNU info manual.

**Comment by unbit** [comment]
This has been discussed lot of times, and the only solution making sense is completely disabling the --help command line option. This could be a solution, but frankly there are literally tons of scripts relying on the --help output to build ad-hoc tools (included autocomplete systems). But let's focus on your proposal: which options you suggest as the most important ones ? because there are so many use cases that some user will be pissed off for sure :) Do not get me wrong, at the current state --help is only useful for scripts (humans can only make jokes on twitter about the number of options in the help ;), so any kind of suggestion is really welcomed.

**Comment by akx** [comment]
> Imagine trying to visually parse that on a VGA TTY.

Is this honestly a real use case these days?

**Comment by xrmx** [comment]
A part of the solution would be to remove some legacy from 2.1 so there's less options available :)

**Comment by johnnybubonic** [comment]
@unbit goodness, who relies on --help output? that's.. interesting. i'd imagine --cgi, --wsgi, --wsgi-file would at least be in them since the documentation mentions them no less than a dozen times. i know how focused you are on backwards-compatibility though, so if there are people actually parsing --help output (seriously, though, i'm...almost impressed) then i'd understand why you'd be hesitant to change them. i think a better structured documentation system *available*, though, like a man page would definitely help.

@akx i can tell you don't spend much time in datacenters. yes, this is a real use case these days. most crash carts/rack terminals still don't have anything but VGA or DVI, and most of the boxes don't have anything but VGA. then again, i don't know why you'd need a 4K LCD HDMI display on a system that only runs a framebuffer console output. chances are if you find a box with an off-board GPU in a rack, it's being used for crunching/rendering, not display.

@xrmx i'm not sure if this was tongue-in-cheek or not, but my understanding is a lot of them already are being planned for removal in 2.1 already?

**Comment by unbit** [comment]
@johnnybubonic another proposal is to have a new flag for options OPT_SHOW_IN_HELP that will allow to the option to be shown in the --help houtput (to see the others you need to use --help-full).

In this way plugin developers have to choose if an option has "the right" to appear in the --help

**Comment by akx** [comment]
@johnnybubonic You're correct that I don't spend any time in data centers. However, I'm tempted to say that if you need to (re)configure uWSGI using a crash cart, something's terribly wrong already and it's a corner case we shouldn't worry about. :D

**Comment by johnnybubonic** [comment]
@unbit perhaps, seems a little ungraceful to me... i can see all plugin developers just opting to be included in --help, so same issue.

[the GNU standards](https://www.gnu.org/prep/standards/standards.html#g_t_002d_002dhelp) provide a brief but, in my opinion, good guideline for -h/--help output (emphasis added):

> The standard --help option should **output _brief_ documentation** for how to invoke the program, on standard output, then exit successfully. Other options and arguments should be ignored once this is seen, and the program should not perform its normal function.
> 
> Near the end of the ‘--help’ option’s output, please place lines giving the email address for bug reports, the package’s home page (normally ‘http://www.gnu.org/software/pkg’, **and the general page for help using GNU programs**. (...)

If the output of --help could be limited to 40-60 lines, I think that'd be a big help. man pages and info pages are intended to provide in-depth coverage of all options, --help is intended for usage info and only the most common usage (or as a reference to further documentation if necessary) as other displays (man pages, info pages, hell- even an ASCII README.txt in /usr/share/doc/uwsgi for instance) paginate and flow the text a lot better, which also lead to allowing more in-depth documentation of each specific option. I think putting _all_ plugin options in --help-full and only core options in --help would be the best approach, if possible.

@akx I don't think you're in any position to determine what a corner case is if you aren't the one deploying in a production environment. uWSGI claims to be friendly to sysadmins. To be friendly to sysadmins, you must consider their needs as not corner cases. Sorry, but not all of us deploy in "the cloud", nor do we want to, and yes, you'd be surprised just how often software needs to be reconfigured at the TTY. Unless you have actual constructive input on this issue instead of trying to trample over a situation you aren't put in, I think i'm done interacting with you.

**Comment by akx** [comment]
@johnnybubonic I apologize for my tone there. Admittedly it wasn't the most constructive.

**Comment by johnnybubonic** [comment]
@akx no harm, no foul. :) we all learn a little bit of new perspective every day.
